core/unix/loader.c

```diff
@@ -741,17 +741,55 @@ privload_search_rpath(privmod_t *mod, bool runpath, const char *name,
                     len = sep - list;
                 /* support $ORIGIN expansion to lib's current directory */
                 origin = strstr(list, RPATH_ORIGIN);
+#    ifdef CLIENT_INTERFACE
+                char path[MAXIMUM_PATH];
+#    endif
                 if (origin != NULL && origin < list + len) {
                     size_t pre_len = origin - list;
+#    ifdef CLIENT_INTERFACE
+                    snprintf(path, MAXIMUM_PATH, "%.*s%.*s%.*s", pre_len, list,
+                             moddir_len, mod->path,
+                             /* the '/' should already be here */
+                             len - strlen(RPATH_ORIGIN) - pre_len,
+                             origin + strlen(RPATH_ORIGIN));
```

Comment:
Let's avoid duplicating this complex snprintf.  Expand RPATH_ORIGIN just once, please.

Reply:
That's going to produce more code though by having to snprintf differently in the #else of !defined(CLIENT_INTERFACE). Sure?

Reply:
I don't see why it would be different: all the code would use the same base path which would not be in any ifdef.

Reply:
Done. That's better indeed, I removed the ifdefs.